mne/stats/tests/test_cluster_level.py

```diff
@@ -911,121 +911,77 @@ def test_new_cluster_api(Inst):
-        ] += 2
-        inst2 = Inst(
-            data=data_tfr2, info=info, times=np.arange(n_times), freqs=np.arange(n_freq)
-        )
+    rng = np.random.default_rng(seed=8675309)
+    is_epo = GetEpochsMixin in Inst.__mro__
+    is_tfr = BaseTFR in Inst.__mro__
 
-    elif Inst == AverageTFRArray:
-        # Create random data for AverageTFRArray
-        data_tfr1 = np.random.randn(n_chan, n_freq, n_times)
-        data_tfr2 = np.random.randn(n_chan, n_freq, n_times)
-        inst1 = Inst(
-            data=data_tfr1, info=info, times=np.arange(n_times), freqs=np.arange(n_freq)
-        )
-        inst2 = Inst(
-            data=data_tfr2, info=info, times=np.arange(n_times), freqs=np.arange(n_freq)
-        )
-        data_tfr2 = inst2.data.copy()
-        data_tfr2[
-            region_start:region_end, freq_start:freq_end, time_start:time_end
-        ] += 2
-        inst2 = Inst(
-            data=data_tfr2, info=info, times=np.arange(n_times), freqs=np.arange(n_freq)
-        )
+    n_epo, n_chan, n_freq, n_times = 6, 3, 4, 5
 
-    if Inst == EvokedArray or Inst == AverageTFRArray:
-        # Generate random noise
-        noise = np.random.normal(loc=0, scale=0.1, size=inst1.data.shape)
-        # add noise to the data of the second subject
-        inst1_n = inst1.copy()
-        inst1_n.data = inst1.data + noise
-        inst2_n = inst2.copy()
-        inst2_n.data = inst2.data + noise
-        data = [inst1, inst2, inst1_n, inst2_n]
-        conds = ["a", "b"] * n_subs
+    # prepare the dimensions of the simulated data, then simulate
+    size = (n_chan,)
+    if is_epo:
+        size = (n_epo, *size)
+    if is_tfr:
+        size = (*size, n_freq)
+    size = (*size, n_times)
+    data = rng.normal(size=size)
+
+    # construct the instance
+    info = create_info(ch_names=n_chan, sfreq=1000, ch_types="eeg")
+    kw = dict(times=np.arange(n_times), freqs=np.arange(n_freq)) if is_tfr else dict()
+    cond_a = Inst(data=data, info=info, **kw)
+    cond_b = cond_a.copy()
+    # introduce a significant difference in a specific region, time, and frequency
+    ch_start, ch_end = 0, 2  # 2 channels
+    t_start, t_end = 2, 4  # 2 times
+    f_start, f_end = 2, 4  # 2 freqs
+    if is_tfr:
+        cond_b._data[..., ch_start:ch_end, f_start:f_end, t_start:t_end] += 2
     else:
-        data = [inst1, inst2]
+        cond_b._data[..., ch_start:ch_end, t_start:t_end] += 2
+    # for Evokeds/AverageTFRs, we create fake "subjects" as our observations within each
+    # condition. We add a bit of noise while we do so.
+    if not is_epo:
+        insts = list()
+        for cond in cond_a, cond_b:
+            for _n in range(n_epo):
+                if not _n:
+                    insts.append(cond)
+                    continue
+                _cond = cond.copy()
+                _cond.data += rng.normal(scale=0.1, size=_cond.data.shape)
+                insts.append(_cond)
```

Comment:
here, `if not _n` means "if n is zero (the first iteration of the loop), just add `cond` to the list. Otherwise, make a copy of `cond`, add noise to it, then add it to the list"